core/src/dird/ndmp_dma_restore_NDMP_NATIVE.cc

```diff
@@ -174,7 +174,7 @@ int SetFilesToRestoreNdmpNative(JobControlRecord* jcr,
      * Restoring a whole directory using this mechanism is much more efficient
      * than creating an namelist entry for every single file and directory below
      * the selected one. */
-    if (node->extract_dir || node->extract) {
+    if (node->extract) {
```

Comment:
extract_dir was a bit confusing.  It sometimes meant that there was an entry inside the sub tree that needed to be extracted, and sometimes it meant that the directory itself needed to be extracted.

This was now seperated more cleanly, so that only `->extract` means that the node should be extracted and `extract_descendant` which always means that there is an entry inside the sub tree that needs extracting.